src/pages/RecipeDetail/RecipeDetail.tsx

```diff
@@ -0,0 +1,36 @@
+import { useParams } from "react-router-dom";
+import { useRecipeQuery } from "./hooks";
+import { InvalidRequestError, NotFoundError } from "./exceptions";
+
+type RecipeParams = {
+  recipeId: string;
+};
+
+const RecipeDetail = () => {
+  const { recipeId } = useParams<RecipeParams>();
+
+  const { recipe, isLoading, isError, error } = useRecipeQuery(recipeId);
+
+  if (isLoading) {
+    return <>Loading...</>;
+  }
+
+  if (isError) {
+    if (error instanceof NotFoundError) {
+      return <>No recipe exists for this ID</>;
+    }
+    if (error instanceof InvalidRequestError) {
+      return <>This ID is in the wrong format</>;
+    }
+    return <>Could not load recipe, please try again later</>;
```

Comment:
Aware these error messages are not the most user friendly, in production I'd hope we have guidance on how best to write these errors, but I guess they'd also be localised.